Analytics/Tests/AnalyticsFactoryTests.swift

```diff
@@ -33,11 +33,20 @@ final class AnalyticsFactoryTests: XCTestCase {
         let module = AnalyticsFactory.makeAnalyticsModule(
             dispatcher: logEventSpy,
             reachabilityHelper: mockReachabilityChecker,
-            deviceRegisteredValue: ReadOnlyCurrentValue(from: .init(true)))
+            deviceRegistering: MockDeviceRegistering())
         module.logEvent(testEvent) { _ in
             expectation.fulfill()
         }
         wait(for: [expectation], timeout: 0.01)
         XCTAssertEqual(logEventSpy.eventsLogged.first, testEvent)
     }
+
+    private final class MockDeviceRegistering: DeviceRegistering {
```

Comment:
Can this class be added to a helper file so that it can be shared? Or do you think it's better to have them as private in each class?

Reply:
No objection for me. I've changed it to an internal class